tests/language/constant_evaluation.cpp

```diff
@@ -0,0 +1,59 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Copyright (c) 2024 The Khronos Group Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+*******************************************************************************/
+
+//  Check that constant evaluated code inside a kernel with some usually
+//  forbidden constructs works.
+
+#include "../common/get_cts_object.h"
+#include "catch2/catch_test_macros.hpp"
+
+namespace language::constant_evaluation {
+
+// A very inefficient way to compute Fibonacci's series using recursion which is
+// forbidden inside non constant-evaluated kernel code/
+auto constexpr f(int v) {
+  if (v < 2) return v;
+  return f(v - 1) + f(v - 2);
+}
+
+// This corresponds to clarification introduced with
+// https://github.com/KhronosGroup/SYCL-Docs/pull/388
+TEST_CASE("constant evaluated code works inside kernel", "[language]") {
```

Comment:
@keryell, please, disable this test case for DPC++ implementation.

I see following errors in CI:
```
/__w/SYCL-CTS/SYCL-CTS/tests/language/constant_evaluation.cpp:33:10: error: SYCL kernel cannot call a recursive function
   33 |   return f(v - 1) + f(v - 2);
      |          ^
/__w/SYCL-CTS/SYCL-CTS/tests/language/constant_evaluation.cpp:31:1: note: function implemented using recursion declared here
   31 | auto constexpr f(int v) {
      | ^
/__w/SYCL-CTS/SYCL-CTS/tests/language/constant_evaluation.cpp:33:21: error: SYCL kernel cannot call a recursive function
   33 |   return f(v - 1) + f(v - 2);
      |                     ^
/__w/SYCL-CTS/SYCL-CTS/tests/language/constant_evaluation.cpp:31:1: note: function implemented using recursion declared here
   31 | auto constexpr f(int v) {
      | ^
2 errors generated.
```

Reply:
https://github.com/KhronosGroup/SYCL-CTS/pull/987 version bump should hopefully address this issue.